### PR TITLE
[oneDNN] Enable fp32 & bf16 Einsum for CPU by default

### PR DESCRIPTION
**Pull request by gaurides**
1. Adding support for MklEinsum by default ie. without #ifdef ENABLE_MKL.
So, with oneDNN enabled, it will use MklEisum for better performance.
2. Also, adding Einsum to the auto_mixed_precision ALLOWLIST so that this op can run in BF16.

### CODE REVIEW

**Comment by gbaned** [comment]
Hi @ezhulenev Can you please review this PR ? Thank you!

**Review by penpornk — APPROVED** [review]
Thank you for the PR and sorry for the delay!

**Review by penpornk — CHANGES_REQUESTED** [review]
It seems that [ARM CI failure](https://github.com/tensorflow/tensorflow/actions/runs/2828843490/jobs/4472154789) could be related to this PR:
```
//bazel_pip/tensorflow/python/ops/parallel_for:math_test_cpu             FAILED in 3 out of 7 in 91.7s
//bazel_pip/tensorflow/python/ops/parallel_for:math_test_gpu             FAILED in 3 out of 7 in 54.6s
//bazel_pip/tensorflow/python:special_math_ops_test_cpu                  FAILED in 6 out of 15 in 33.5s
//bazel_pip/tensorflow/python:special_math_ops_test_gpu                  FAILED in 3 out of 12 in 33.0s
//bazel_pip/tensorflow/python/kernel_tests/linalg:einsum_op_test_cpu     FAILED in 12 out of 12 in 98.6s
//bazel_pip/tensorflow/python/kernel_tests/linalg:einsum_op_test_gpu     FAILED in 12 out of 12 in 77.1s
```
Maybe guard the change to avoid affecting the aarch64 build? (This CI tests with `TF_ENABLE_ONEDNN_OPTS=1`.)

[Windows Bazel GPU](https://source.cloud.google.com/results/invocations/f8697609-2e62-4cf0-a8ac-59a8cc3f1d22/targets/%2F%2Ftensorflow%2Ftools%2Fci_build%2Fbuilds:gen_win_out/log) and [MacOS CPU Python3](https://source.cloud.google.com/results/invocations/315b3c6c-e5e9-4f1d-beef-0bd11c437fa7/targets/tensorflow%2Frel%2Fmacos%2Fgithub_presubmit%2Fcpu_py39_full/log) failures could be unrelated. I'll rerun the CIs to deflake.

**Comment by penpornk** [comment]
cc: @nSircombe @milpuz01 FYI.

**Comment by gaurides** [comment]
> Maybe guard the change to avoid affecting the aarch64 build? (This CI tests with `TF_ENABLE_ONEDNN_OPTS=1`.)
@penpornk : Thanks for your review and the suggestion. I updated the PR.

**Comment by gaurides** [comment]
@penpornk : The ARM CI build failed. I have added the condition, so now it should be safe to merge. Please can you take a relook? Thanks.

**Comment by gaurides** [comment]
[@elfringham](https://github.com/elfringham) : do you know why ARM CI failed for this PR too?

File: tensorflow/core/grappler/optimizers/auto_mixed_precision_lists.h [hunk 1]
```diff
@@ -375,7 +375,8 @@ class AutoMixedPrecisionListsMkl : public AutoMixedPrecisionLists {
                                      "DepthwiseConv2dNativeBackpropInput",
                                      "MatMul",
                                      "BatchMatMul",
-                                     "BatchMatMulV2"};
+                                     "BatchMatMulV2",
+                                     "Einsum"};
```
Comment (penpornk) [inline tensorflow/core/grappler/optimizers/auto_mixed_precision_lists.h:379]: Do we need to wrap this with `#ifndef DNNL_AARCH64_USE_ACL` too?

**Review by penpornk — CHANGES_REQUESTED** [review]

File: tensorflow/core/grappler/optimizers/auto_mixed_precision_lists.h (hunk 1 above)
Comment (penpornk) [reply tensorflow/core/grappler/optimizers/auto_mixed_precision_lists.h:379]: Never mind. I made this change internally in the [merge commit](https://github.com/tensorflow/tensorflow/commit/3f56300c412b4b113c6a551ad1459035671900af).

Comment (gaurides) [reply tensorflow/core/grappler/optimizers/auto_mixed_precision_lists.h:379]: thanks @penpornk for your help.

**Pull request merged**